### PR TITLE
Feature/dialog system

### PR DESCRIPTION
**Pull request by Heliarco**

### CODE REVIEW

File: Character/Action_Dash.gd [hunk 1]
```diff
@@ -0,0 +1,16 @@
+extends Node
```
Comment (mmarramarco) [inline Character/Action_Dash.gd]: delete this whole file? Why is there GD code here x)

File: UI/DialogSystem/DialogSystem.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+public class DialogSystem : CanvasLayer 
+{
+    private RichTextLabel _textLabel;
+    private IEnumerator<float> _coRoutine;
+    private Queue<string> _dialogQueue = new Queue<string>();
+
```
Comment (mmarramarco) [inline UI/DialogSystem/DialogSystem.cs]: can you remove those extra empty lines? they're not necessary and it just makes files bigger and less easy to read

File: UI/DialogSystem/DialogSystem.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    public void PlayDialog(string message)
+    {    
+        _dialogQueue.Enqueue(message);
+    }
+
+    // The actual UI Co-routine, return value indicates desired delay between actions in seconds
+    private IEnumerable<float> InnerProcess()
+    {
+        while(true) {
```
Comment (mmarramarco) [inline UI/DialogSystem/DialogSystem.cs]: C# conventions, { on a new line.
Also, is the while (true) relly necessary ^^' i'm really not confident reading this kind of code...

File: UI/DialogSystem/DialogSystem.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
+using Godot;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+public class DialogSystem : CanvasLayer
```
Comment (mmarramarco) [inline UI/DialogSystem/DialogSystem.cs:6]: xmldoc to explain how to use this dialog system class would be cool

File: UI/DialogSystem/Models/SimpleDialog.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+using Godot;
+using System;
+using System.Collections.Generic;
+
+public class SimpleDialog : Resource
```
Comment (mmarramarco) [inline UI/DialogSystem/Models/SimpleDialog.cs:5]: each class should be its own file.

**Review by mmarramarco — CHANGES_REQUESTED** [review]

File: Character/Action_Dash.gd (hunk 1 above)
Comment (Heliarco) [reply Character/Action_Dash.gd]: Really good point, don't even know where that came from >_> consider it gone

File: UI/DialogSystem/DialogSystem.cs (hunk 2 above)
Comment (Heliarco) [reply UI/DialogSystem/DialogSystem.cs]: Indeed (Another good reason to get autolinting up and running :D)

File: UI/DialogSystem/DialogSystem.cs (hunk 4 above)
Comment (Heliarco) [reply UI/DialogSystem/DialogSystem.cs:6]: Will come when its ready for use, for now it is not yet usable

File: UI/DialogSystem/DialogSystem.cs (hunk 3 above)
Comment (Heliarco) [reply UI/DialogSystem/DialogSystem.cs]: Yes it is. 
Its a c# Iterator block "https://www.c-sharpcorner.com/UploadFile/5ef30d/understanding-yield-return-in-C-Sharp/"
It autogenerates a statemachine, and allows me to have a method where i can say "run until next step"
Really usefull in long running background operations (Such as having text slowly being typed)

**Review by mmarramarco — APPROVED** [review]

**Pull request merged**